types/parse/index.d.ts

```diff
@@ -458,9 +465,9 @@ declare namespace Parse {
      *   }
      * });</pre></p>
      */
-    class Query<T extends Object = Object> extends BaseObject {
+    class Query<T = Object, K extends Unobject<T> = Unobject<T>> extends BaseObject {
 
-        objectClass: any;
+        objectClass: T;
         className: string;
 
         constructor(objectClass: string);
```

Comment:
@dirkgroenen  The other constructor `constructor(objectClass: new (...args: any[]) => T);` could be `constructor(objectClass: new (...args: any[]) => Object);`. In this way, it is possible to use `new Parse.Query(Parse.Object)`, although it is no more able to infer the generic type from constructor. It would be necessary to do `new Parse.Query<MyObj>(MyObj)` if I have a `class MyObj extends Object<{ foo: string }> {}` for example.

What do you think?

Reply:
Giving up on our inferred type feels a bit like giving up one of the better improvements in this PR. I'm eager on finding the reason, or a better solution, which will solve the problems we're having. 

I have a few ideas which I would like to try before falling back to manually providing our type. I'll try to come back on this thread this week.